### DISCUSSION TITLE
Cooler Balance failed

### DISCUSSION THREAD

**Discussion by kimquililan** [open2c/cooler#371] [Q&A]
Hi! I ran cooler balance with my cool file and it ran okay until it reached an error

Output exceeds the size limit. Open the full output data in a text editor
INFO:cooler.cli.balance:Balancing "INO80_4_FA_FilteredbyDist_ShiftbyOrientation_80_bin20.cool"
INFO:cooler.balance:variance is 340.4289927827929
INFO:cooler.balance:variance is 313.23710687493167
INFO:cooler.balance:variance is 185.8196031817053
INFO:cooler.balance:variance is 333.761573580506
INFO:cooler.balance:variance is 196.25432819720305
INFO:cooler.balance:variance is 358.0699084083024
INFO:cooler.balance:variance is 209.38936590752522
INFO:cooler.balance:variance is 376.30319720607156
INFO:cooler.balance:variance is 220.66785884443226
INFO:cooler.balance:variance is 390.03982936886524
INFO:cooler.balance:variance is 230.22842018905308
INFO:cooler.balance:variance is 401.1281708234125
INFO:cooler.balance:variance is 238.69538102448956
INFO:cooler.balance:variance is 410.96323399634514
INFO:cooler.balance:variance is 246.59651335609055
INFO:cooler.balance:variance is 420.5124418137992
INFO:cooler.balance:variance is 254.30424005915543
INFO:cooler.balance:variance is 430.42442586082234
INFO:cooler.balance:variance is 262.070027289631
INFO:cooler.balance:variance is 441.12642971726603
INFO:cooler.balance:variance is 270.0644332808771
INFO:cooler.balance:variance is 452.89700356277393
INFO:cooler.balance:variance is 278.4066275301745
INFO:cooler.balance:variance is 465.9168997527476
...
  warnings.warn(

ERROR:cooler.cli.balance:Iteration limit reached without convergence
ERROR:cooler.cli.balance:Storing final result. Check log to assess convergence.

When I do cooler dump, the balanced column is empty. Did I do something wrong?

**Comment by agalitsyna** [comment]
Hi, Kim! 
This usually means that the matrix is too sparse for balancing at this resolution, and rows/columns of Hi-C matrix have too little counts for balancing. 
Before confirming that, I would limit the number of iterations for balancing by `--max-iters` and set it to 10, just to see how the maps will look like!
